oonimkall/runner_test.go

```diff
@@ -4,49 +4,60 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"log"
+	"strings"
 	"testing"
 
 	engine "github.com/ooni/probe-engine"
 )
 
 func TestUnitRunnerHasUnsupportedSettings(t *testing.T) {
-	var noFileReport, randomizeInput bool
 	out := make(chan *eventRecord)
 	settings := &settingsRecord{
 		InputFilepaths: []string{"foo"},
+		Name:           "example",
 		Options: settingsOptions{
 			Backend:          "foo",
 			CABundlePath:     "foo",
 			GeoIPASNPath:     "foo",
 			GeoIPCountryPath: "foo",
-			NoFileReport:     &noFileReport,
+			NoFileReport:     false,
 			ProbeASN:         "AS0",
 			ProbeCC:          "ZZ",
 			ProbeIP:          "127.0.0.1",
 			ProbeNetworkName: "XXX",
-			RandomizeInput:   &randomizeInput,
+			RandomizeInput:   true,
 		},
-		OutputFilePath: "foo",
+		OutputFilepath: "foo",
 	}
-	numseen := make(chan int)
 	go func() {
```

Comment:
The implementation of this test was unnecessarily complex and ineffective. Now it should be more strict and easier.